creator-node/src/blacklistManager.js

```diff
@@ -8,103 +7,201 @@ const REDIS_SET_BLACKLIST_USERID_KEY = 'SET.BLACKLIST.USERID'
-
-  const trackIds = new Set(trackBlacklist)
-
-  // Fetch all tracks created by users in userBlacklist
-  let trackBlockchainIds = []
-  if (userBlacklist.length > 0) {
-    trackBlockchainIds = (await models.sequelize.query(
-      'select "blockchainId" from "Tracks" where "cnodeUserUUID" in (' +
-        'select "cnodeUserUUID" from "AudiusUsers" where "blockchainId" in (:userBlacklist)' +
-      ');'
-      , { replacements: { userBlacklist } }
-    ))[0]
-  }
-  if (trackBlockchainIds) {
-    for (const trackObj of trackBlockchainIds) {
-      if (trackObj.blockchainId) {
-        trackIds.add(trackObj.blockchainId)
-      }
+    // Dedupe trackIds
+    const trackIds = new Set(trackIdsToBlacklist)
+
+    // Retrieves CIDs from deduped trackIds
+    const segmentCIDsToBlacklist = await this.getCIDsFromTrackIds([...trackIds])
+
+    try {
+      await this.addTrackIdsToRedis(trackIdsToBlacklist)
+      await this.addUserIdsToRedis(userIdsToBlacklist)
+      await this.addCIDsToRedis(segmentCIDsToBlacklist)
+    } catch (e) {
+      throw new Error(`Failed to add to blacklist: ${e}`)
     }
   }
 
-  return { trackIdsToBlacklist: [...trackIds], userIdsToBlacklist: userBlacklist }
-}
+  /**
+  * Given trackIds and userIds to remove from blacklist, fetch all segmentCIDs.
+  * Also remove the trackIds, userIds, and segmentCIDs from redis blacklist sets to prevent future interaction.
+  */
+  static async remove (trackIdsToRemove = [], userIdsToRemove = []) {
+    // Get tracks from param and by parsing through user tracks
+    const tracks = await this.getTracksFromUsers(userIdsToRemove)
+    trackIdsToRemove = [...tracks.map(track => track.blockchainId), ...trackIdsToRemove]
+
+    // Dedupe trackIds
+    let trackIds = new Set(trackIdsToRemove)
 
-/**
- * Given trackIds and userIds to blacklist, fetch all segmentCIDs and unpin from IPFS.
- * Also add all trackIds, userIds, and segmentCIDs to redis blacklist sets to prevent future interaction.
- */
-async function _processBlacklist (ipfs, trackIdsToBlacklist, userIdsToBlacklist) {
-  const tracks = await models.Track.findAll({ where: { blockchainId: trackIdsToBlacklist } })
+    // Retrieves CIDs from deduped trackIds
+    const segmentCIDsToRemove = await this.getCIDsFromTrackIds([...trackIds])
+
+    try {
+      await this.removeTrackIdsFromRedis(trackIdsToRemove)
+      await this.removeUserIdsFromRedis(userIdsToRemove)
+      await this.removeCIDsFromRedis(segmentCIDsToRemove)
+    } catch (e) {
+      throw new Error(`Failed to remove from blacklist: ${e}`)
+    }
+  }
 
-  let segmentCIDsToBlacklist = new Set()
+  /**
+   * Retrieves track objects from specified users
+   * @param {[int]} userIdsBlacklist
+   */
+  static async getTracksFromUsers (userIdsBlacklist) {
+    let tracks = []
+    if (userIdsBlacklist.length > 0) {
+      // ? this actually returns the entire track and not just blockchainId
+      tracks = (await models.sequelize.query(
```

Comment:
any particular reason you didn't use sequelize here? that might solve your select issue (?)

Reply:
ah this was already here. I just moved the logic around